code-studio/src/js/initApp/initApp.js

```diff
@@ -169,8 +169,8 @@ window.apps = {
         // If we're readOnly, source hasn't changed at all
         source = Blockly.mainBlockSpace.isReadOnly() ? currentLevelSource :
           Blockly.Xml.domToText(Blockly.Xml.blockSpaceToDom(Blockly.mainBlockSpace));
-      } else {
-        source = window.Applab && Applab.getCode();
+      } else if (appOptions.getCode) {
+        source = appOptions.getCode();
```

Comment:
Oops!  Sorry to hear that.  The shared directory is still around, but the shared/js package moved into code-studio.  Take a look at [the README for code-studio](https://github.com/code-dot-org/code-dot-org/tree/staging/code-studio) or ask me if you have any questions.